src/super_gradients/training/metrics/segmentation_metrics.py

```diff
@@ -77,6 +124,26 @@ def intersection_and_union(im_pred, im_lab, num_class):
     return area_inter, area_union
 
 
+def _preprocess_segmentation_inputs(preds, target: torch.Tensor,
```

Comment:
We are getting more and more of these for each new scenario...
My suggestion- lets make a callable class (named MetricsArgsPrepFn or something similar) that we can just pass to our Metrics objects, and would apply some transformations to input and targets before applying the metrics update.
In this case wed inherit from it and construct a class named PreprocessSegmentationMetricsArgsFn that applies the below code accordingly.